### PR TITLE
Fix intermittent test failures in TextSuite

### LINKED ISSUE #1950: TextSpec failure on CI build
```scala
 [info]   - 1 byte sequences *** FAILED *** (66 milliseconds)
[info]     GeneratorDrivenPropertyCheckFailedException was thrown during property evaluation. (TextSpec.scala:78)
[info]       Falsified after 4 successful property evaluations.
[info]       Message: List() did not equal List("﻿")
[info]       Location: (TextSpec.scala:79)
[info]       Occurred when passed generated values (
[info]         "﻿"
[info]       )
[info]       Init Seed: 1594344922141
```

### PR DESCRIPTION
**Pull request by mpilquist**
Fixes #1950

### CODE REVIEW

File: core/shared/src/test/scala/fs2/TextSuite.scala [hunk 1]
```diff
@@ -136,13 +139,13 @@ class TextSuite extends Fs2Suite {
     group("handles byte order mark") {
       val bom = Chunk[Byte](0xef.toByte, 0xbb.toByte, 0xbf.toByte)
       property("single chunk") {
-        forAll { (s: String) =>
+        forAll(genStringNoBom) { (s: String) =>
```
Comment (djspiewak) [inline core/shared/src/test/scala/fs2/TextSuite.scala:142]: This seems like a very ironic diff.

Comment (mpilquist) [reply core/shared/src/test/scala/fs2/TextSuite.scala:142]: @djspiewak Took me a minute to catch on.

BTW, I'm not sure FS2 should really strip the byte order mark -- if FS2 just passed it along, these tests would all have passed. Looking at https://github.com/typelevel/fs2/issues/1484, we never really discussed why removing it is right behavior. Here's a quote from wikipedia: "The standard also does not recommend removing a BOM when it is there, so that round-tripping between encodings does not lose information, and so that code that relies on it continues to work." (https://en.wikipedia.org/wiki/Byte_order_mark). Thoughts?

Comment (djspiewak) [reply core/shared/src/test/scala/fs2/TextSuite.scala:142]: IMO if the conversion is to `String`, it's better to strip the BOM. `String` itself already internally establishes all of the invariants that are communicated via the BOM (codec and endianness), so it's just redundant information which messes up offsets invisibly. However, as long as things are in byte-form, the BOM needs to be preserved because we have no other way of signalling those things.

In other words, I *think* I believe that `utf8decode` is doing the right thing. A `Pipe[F, Byte, String]` should strip the BOM, while a `Pipe[F, Byte, Byte]` (or analogous) should not.
